dbt/include/athena/macros/materializations/models/table/create_table_as.sql

```diff
@@ -9,17 +9,41 @@
   {%- set s3_data_dir = config.get('s3_data_dir', default=target.s3_data_dir) -%}
   {%- set s3_data_naming = config.get('s3_data_naming', default=target.s3_data_naming) -%}
 
+  {%- if format == 'iceberg' -%}
+    {%- set location_property = 'location' -%}
+    {%- set partition_property = 'partitioning' -%}
+    {%- if bucketed_by is not none or bucket_count is not none -%}
+      {%- set error_bucket_iceberg -%}
+      bucketed_by or bucket_count cannot be used with Iceberg tables. You have to use the bucket function
+      when partitioning
+      {%- endset -%}
+      {% do exceptions.raise_compiler_error(error_bucket_iceberg) %}
+    {%- endif -%}
+    {%- if s3_data_naming in ['table', 'table_schema'] -%}
+      {%- set error_s3_data_naming_iceberg -%}
+        S3 data naming must be set to a strategy that enforces unique table locations when creating Iceberg table
+        using CTAS because we are using the RENAME feature to provide near-zero downtime.
+      {%- endset -%}
+      {% do exceptions.raise_compiler_error(error_s3_data_naming_iceberg) %}
+    {%- endif -%}
+  {%- else -%}
+    {%- set location_property = 'external_location' -%}
+    {%- set partition_property = 'partitioned_by' -%}
+  {%- endif %}
+
   create table
     {{ relation }}
 
     with (
-      {%- if external_location is not none and not temporary %}
-        external_location='{{ external_location }}',
+        table_type={%- if format == 'iceberg' -%}'iceberg'{%- else -%}'hive'{%- endif %},
+        is_external={%- if format == 'iceberg' -%}false{%- else -%}true{%- endif %},
+      {% if external_location is not none and not temporary %}
+        {{ location_property }}='{{ external_location }}',
       {%- else -%}
-        external_location='{{ adapter.s3_table_location(s3_data_dir, s3_data_naming, relation.schema, relation.identifier) }}',
+        {{ location_property }}='{{ adapter.s3_table_location(s3_data_dir, s3_data_naming, relation.schema, relation.identifier) }}',
```

Comment:
note: revisit the `s3_table_location` function to avoid to use trailing slash in the location, doing so we will avoid issues on dropping the table (there is an underlying bug in athena due to that)!